tck/old-tck/source/src/com/sun/ts/tests/jsf/api/jakarta_faces/application/application/TestServlet.java

```diff
@@ -1467,32 +1468,30 @@ public void applicationGetExpressionFactoryTest(HttpServletRequest request,
 
     // Left in for debugging.
     //
-    // ExpressionFactory controlFactoryTwo =
-    // JspFactory.getDefaultFactory().
-    // getJspApplicationContext(servletContext).getExpressionFactory();
+    // ExpressionFactory controlFactoryTwo = ELManager.getExpressionFactory();
     //
     // ExpressionFactory exprFactoryTwo =
     // application.getExpressionFactory();
     // out.println("*** First Call application.getExpressionFactory() " +
     // "Returns ***: " + exprFactory.toString());
     // out.println("*** Second Call application.getExpressionFactory() " +
     // "Returns ***: " + exprFactoryTwo.toString());
-    // out.println("*** First Call - JspFactory.getDefaultFactory()." +
+    // out.println("*** First Call - ELManager.getExpressionFactory()." +
     // "getJspApplicationContext(servletContext)." +
     // "getExpressionFactory() Returns ***: " +
     // controlFactory.toString());
-    // out.println("*** Second Call - JspFactory.getDefaultFactory()." +
+    // out.println("*** Second Call - ELManager.getExpressionFactory()." +
     // "getJspApplicationContext(servletContext)." +
     // "getExpressionFactory() Returns ***: " +
     // controlFactoryTwo.toString());
 
-    if (!controlFactory.equals(exprFactory)) {
+    if (!exprFactory.toString().equals(controlFactory.toString())) {
       out.println(JSFTestUtil.FAIL + JSFTestUtil.NL
           + "Expected Application.getExpressionFactory to return "
-          + "the same instance as that returned by "
-          + "JspApplicationContext.getExpressionFactory!" + JSFTestUtil.NL
-          + "Found: " + controlFactory.toString() + JSFTestUtil.NL
-          + "Expected: " + exprFactory.toString());
+          + "the same type as that returned by "
```

Comment:
"same type" is probably actually misleading here, considering what toString() actually outputs.

Here's an example of the failure text:
```
[javatest.batch] Expected Application.getExpressionFactory to return the same type as that returned by ELManager.getExpressionFactory()!
[javatest.batch] ELManager.getExpressionFactory: org.apache.el.ExpressionFactoryImpl@df82862f
[javatest.batch] application.getExpressionFactory: org.apache.el.ExpressionFactoryImpl@4d5b0b28
```